File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedMapEntry.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.MapEntryRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.StyledText;
import javafx.scene.Node;

import java.util.ArrayList;
import java.util.List;

import static com.ulyp.ui.util.CssClass.CALL_TREE_PLAIN_TEXT;

public class RenderedMapEntry extends RenderedObject {

    protected RenderedMapEntry(MapEntryRepresentation entry, RenderSettings renderSettings) {
        super(entry.getType());

        List<Node> texts = new ArrayList<>();

        texts.add(RenderedObject.of(entry.getKey(), renderSettings));
        texts.add(StyledText.of(" -> ", CALL_TREE_PLAIN_TEXT));
        texts.add(RenderedObject.of(entry.getValue(), renderSettings));

        super.getChildren().addAll(texts);
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedMapEntry.kt
package com.ulyp.ui.renderers

import com.ulyp.core.printers.MapEntryRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of
import javafx.scene.Node

class RenderedMapEntry(entry: MapEntryRecord, renderSettings: RenderSettings?) : RenderedObject(entry.type) {

    init {
        val texts: MutableList<Node> = ArrayList()
        texts.add(of(entry.key, renderSettings))
        texts.add(of(" -> ", CssClass.CALL_TREE_NODE_SEPARATOR))
        texts.add(of(entry.value, renderSettings))
        super.getChildren().addAll(texts)
    }
}